collections/nemo_asr/nemo_asr/audio_preprocessing.py

```diff
@@ -0,0 +1,470 @@
+# Copyright (C) NVIDIA CORPORATION. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.****
+"""
+This file contains neural modules responsible for preprocessing audio data.
+"""
+__all__ = ['AudioPreprocessor',
+           'AudioToMFCCPreprocessor',
+           'AudioToMelSpectrogramPreprocessor',
+           'AudioToSpectrogramPreprocessor',
+           'MultiplyBatch',
+           'SpectrogramAugmentation']
+
+import math
+import torch
+import torchaudio
+try:
+    from apex import amp
+except AttributeError:
+    print("Unable to import APEX. Mixed precision and distributed training "
+          "will not work.")
+
+from nemo.backends.pytorch import NonTrainableNM
+from nemo.core import Optimization
+from nemo.core.neural_types import *
+from .parts.features import FilterbankFeatures
+from .parts.spectr_augment import SpecAugment, SpecCutout
+
+
+class AudioPreprocessor(NonTrainableNM):
+    """
+    A base class for Neural Modules that performs audio preprocessing,
+    transforming the wav files to features.
+    """
+    @staticmethod
+    def create_ports():
```

Comment:
This class shouldn't have ports

Reply:
Each child class should be returning ports with different tags. As part of nemo_tts, I added MelSpectrogramSignalTag. Please add one for MFCC too.

Reply:
Done